src/lazy/expanded/macro_evaluator.rs

```diff
@@ -499,18 +499,12 @@ impl<'top, D: Decoder> MacroExpansion<'top, D> {
             ExprGroup(expr_group_expansion) => expr_group_expansion.next(context, environment),
             MakeString(make_string_expansion) => make_string_expansion.make_text_value(context),
             MakeSymbol(make_symbol_expansion) => make_symbol_expansion.make_text_value(context),
-            MakeSExp(make_sexp_expansion) => make_sexp_expansion.next(context, environment),
             Annotate(annotate_expansion) => annotate_expansion.next(context, environment),
+            Flatten(flatten_expansion) => flatten_expansion.next(),
             // `none` is trivial and requires no delegation
             None => Ok(MacroExpansionStep::FinalStep(Option::None)),
         }
     }
-
-    // Calculate the next step in this macro expansion without advancing the expansion.
-    pub fn peek_next_step(&self) -> IonResult<MacroExpansionStep<'top, D>> {
-        let mut expansion_copy = *self;
```

Comment:
🪧 The aforementioned use of `Copy`, which would have enabled `peek()`ing at the next step of a macro. This is still possible without `Copy`, but will take a bit more code.